### ISSUE TITLE
`Version.parsed()` incorrectly accepts invalid semver trailing inputs

### ISSUE THREAD

**Issue by zaguzovmaksim0-hue** [objectionary/lints#1450]
## What happens

`Version.parsed()` documents a strict `major.minor.patch` version, and `LtSyntaxVersion` says non-SemVer values are ignored. But `Version.CORE` is anchored only at the beginning:

```java
private static final Pattern CORE = Pattern.compile("^(\\d+)\\.(\\d+)\\.(\\d+)");
```

There is no end condition or validation of what follows the patch component. On current master a same-package harness over the current `Version.java` gives:

```text
1.2.3 -> Optional[1.2.3]
1.2.3-alpha.1 -> Optional[1.2.3]
1.2.3garbage -> Optional[1.2.3]
1.2.3.4 -> Optional[1.2.3]
999.0.0oops -> Optional[999.0.0]
```

A valid prerelease is intentionally compared by its numeric core. Arbitrary text and a fourth numeric component are not valid SemVer but are silently accepted as valid versions.

## Impact

`LtSyntaxVersion` uses `Version.parsed(declared)` to decide whether `+syntax` requires a newer parser. Thus `+syntax 999.0.0oops` may produce a false hard error requiring future EO syntax instead of being ignored as malformed/non-SemVer, contradicting the lint's own documented contract.

## What should happen

Validate the complete spelling while allowing valid SemVer prerelease/build suffixes to be ignored for numeric comparison. Arbitrary trailing text and extra numeric components should return `Optional.empty()`.

A regression should cover `1.2.3garbage` and `1.2.3.4` while preserving `1.2.3-alpha.1`.

I searched existing issues for `syntax-version garbage suffix`, `Version parsed trailing garbage`, `non-SemVer syntax-version accepted`, `syntax version 1.2.3.4`, and `syntax-version regex anchored`; no duplicate was found.

**Comment by 0crat** [comment]
@zaguzovmaksim0-hue Thanks for reporting a new bug! You've earned +12 points. By reporting bugs, you help our project improve its quality. If you find anything else in the repository that [doesn't look](https://www.yegor256.com/2018/02/06/where-to-find-more-bugs.html) as good as you might expect, [do not hesitate](https://www.yegor256.com/2014/04/13/bugs-are-welcome.html) to [report](https://www.yegor256.com/2018/04/24/right-way-to-report-bugs.html) it. Your running score is [+279](https://www.eolang.org/zerocracy/objectionary-vitals.html); don't forget to check your [Zerocracy account](https://www.zerocracy.com/) too).
